bba/bba.go

```diff
@@ -109,26 +111,29 @@ func New(
 
 		auxBroadcasted: cleisthenes.NewBinaryState(),
 
-		bvalRepo:        newBvalReqRepository(),
-		auxRepo:         newAuxReqRepository(),
-		incomingReqRepo: newDefaultIncomingRequestRepository(),
+		bvalRepo:            newBvalReqRepository(),
+		auxRepo:             newAuxReqRepository(),
+		incomingBvalReqRepo: newDefaultIncomingRequestRepository(),
+		incomingAuxReqRepo:  newDefaultIncomingRequestRepository(),
 
 		closeChan: make(chan struct{}, 1),
 
 		// request channel size as n*8, because each node can have maximum 4 rounds
 		// and in each round each node can broadcast at most twice (broadcast BVAL, AUX)
 		// so each node should handle 8 requests per node.
-		reqChan:             make(chan request, n*8),
-		binValueChan:        make(chan struct{}, n),
-		tryoutAgreementChan: make(chan struct{}, n),
-		advanceRoundChan:    make(chan struct{}, n),
+		reqChan: make(chan request, n*8),
+		// below channel size as n*4, because each node can have maximum 4 rounds
+		binValueChan:        make(chan uint64, n*4),
+		tryoutAgreementChan: make(chan uint64, n*4),
+		advanceRoundChan:    make(chan uint64, n*4),
 
 		broadcaster:   broadcaster,
-		coinGenerator: coinGenerator,
+		coinGenerator: mock.NewCoinGenerator(cleisthenes.Coin(cleisthenes.One)),
```

Comment:
why do not DI coin generator?

Reply:
1. only BBA use `common coin`
2. each BBA instance should has it's own coin generator